pkg/diagnose/cni.go

```diff
@@ -73,9 +73,9 @@ func CNIConfig(clusterInfo *cluster.Info, status reporter.Interface) bool {
 	}
 
 	if !isSupportedPlugin {
-		status.Failure("The detected CNI network plugin (%q) is not supported by Submariner."+
-			" Supported network plugins: %v\n", clusterInfo.Submariner.Status.NetworkPlugin, supportedNetworkPlugins)
-		return false
+		return status.Error(errors.New("Unsupported CNI plugin"),
+			"The detected CNI network plugin (%q) is not supported by Submariner."+
+				" Supported network plugins: %v\n", clusterInfo.Submariner.Status.NetworkPlugin, supportedNetworkPlugins)
```

Comment:
The detailed info is in the passed message so appending the error string is redundant and looks a bit awkward.

```suggestion
		status.Failure("The detected CNI network plugin (%q) is not supported by Submariner."+
				" Supported network plugins: %v\n", clusterInfo.Submariner.Status.NetworkPlugin, supportedNetworkPlugins)
		return errors.New("unsupported CNI plugin")
```